src/components/pages/Wallet.tsx

```diff
@@ -310,17 +430,61 @@ const mapStateToProps: MapStateToProps<
   const loggedInMember = getLoggedInMember(state);
   if (!loggedInMember) {
     // TODO: gracefully deal with this situation.
-    console.error("Member not logged in, should not have gotten here.");
-    return {} as StateProps;
+    throw Error("Member not logged in, should not have gotten here.");
   }
+  const loggedInMemberId = loggedInMember.get("memberId");
   return {
     loggedInMember,
-    mintableAmount: getMintableAmount(state, loggedInMember),
-    unclaimedReferralIds: getUnclaimedReferrals(
+    mintableBasicIncome: getMintableBasicIncomeAmount(state, loggedInMemberId),
+    unclaimedReferralIds: getUnclaimedReferrals(state, loggedInMemberId),
+    mintableInvitedBonus: getMintableInvitedBonus(state, loggedInMember),
+    mintApiCallStatus: getStatusOfApiCall(
       state,
-      loggedInMember.get("memberId")
+      ApiEndpointName.MINT,
+      loggedInMemberId
     )
   };
 };
 
-export const Wallet = connect(mapStateToProps)(WalletView);
+const mergeProps: MergeProps<
+  StateProps,
+  DispatchProps,
+  OwnProps,
+  MergedProps
+> = (stateProps, dispatchProps, ownProps) => {
+  const {
+    loggedInMember,
+    mintableBasicIncome,
+    mintableInvitedBonus
+  } = stateProps;
+  var mintActions = [] as MintArgs[];
+  if (mintableBasicIncome && mintableBasicIncome.gt(0)) {
```

Comment:
nit: given the typing as `Big` instead of `Big || undefined`, I think we can get rid of the existence check